task/sast-coverity-check/0.1/sast-coverity-check.yaml

```diff
@@ -0,0 +1,243 @@
+      default: cov-license
+    - name: AUTH_TOKEN_COVERITY_IMAGE
+      description: Name of secret which contains the authentication token for pulling the Coverity image.
+      default: "auth-token-coverity-image"
+    - name: IMP_FINDINGS_ONLY
+      type: string
+      description: Report only important findings. Default is true. To report all findings, specify "false"
+      default: "true"
+    - name: KFP_GIT_URL
+      type: string
+      description: URL from repository to download known false positives files
+      default: "" # FIXME: Red Hat internal projects will default to https://gitlab.cee.redhat.com/osh/known-false-positives.git when KONFLUX-4530 is resolved
+    - name: PROJECT_NVR
+      type: string
+      description: Name-Version-Release (NVR) of the scanned project, used to find path exclusions (it is optional)
+      default: ""
+    - name: RECORD_EXCLUDED
+      type: string
+      description: Write excluded records in file. Useful for debugging (it is optional).
+      default: "false"
+  volumes:
+    - name: cov-license
+      secret:
+        secretName: $(params.COV_LICENSE)
+        optional: false
+    - name: auth-token-coverity-image
+      secret:
+        secretName: $(params.AUTH_TOKEN_COVERITY_IMAGE)
+        optional: false
+  steps:
+    - name: sast-coverity-check
+      onError: stopAndFail
+    # TODO: Change image
+      #image: $(steps.secrets-check.results.image)
+      image: quay.io/redhat-user-workloads/sast-tenant/sast-scanner/coverity@sha256:1e6d26ef45da80128c6bbb4f29a29ac3e547e934e533cc5c1dd37e40439269fe
+      computeResources:
+        requests:
+          memory: "16Gi"
+          cpu: "8"
+        limits:
+          memory: "32Gi"
+          cpu: "16"
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+        - name: cov-license
+          mountPath: "/etc/secrets/cov"
+          readOnly: true
+      env:
+        - name: COV_ANALYZE_ARGS
+          value: $(params.COV_ANALYZE_ARGS)
+        - name: COV_CAPTURE_ARGS
+          value: $(params.COV_CAPTURE_ARGS)
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: COV_LICENSE
+          value: $(params.COV_LICENSE)
+        - name: IMP_FINDINGS_ONLY
+          value: $(params.IMP_FINDINGS_ONLY)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+      script: |
+        #!/usr/bin/env bash
+        set -eo pipefail
+        . /usr/local/share/konflux-test/utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+        
+        echo 'Starting Coverity buildless scan'
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)
+        COVERITY_DIR=/tmp/cov-scan/cov
```

Comment:
I asked the Konflux team about this.
Waiting for a response....

Reply:
Just spoke with Konflux infra team:

> default [storage] is 1GB
> 2. increase is as much as you need, I do not know what is the limit

Do we have an estimation on how much show I increase it? Or should I take an arbitrary value (such as 10Gb)

Related docs: https://konflux-ci.dev/docs/troubleshooting/#no-space-left-on-device

Reply:
`1 GB` is definitely not going to be enough for certain scans.  I roughly remember there was (at least on some mount points) more space that last time I was debugging the buildless capture in the Konflux environment.  We may try 16 GB initially but I suspect it will not be enough in some cases anyway.

Reply:
Sure. This will need a bigger discussion with Konflux team.
The `pipelinerun` file is automatically generated by the build service: https://github.com/konflux-ci/build-service/blob/b2beb5c352b9d8c9992932ad33a16096deafa629/controllers/component_build_controller_pipeline.go#L416 
Right now, the default is 1Gi, being defined by https://github.com/konflux-ci/build-service/blob/b2beb5c352b9d8c9992932ad33a16096deafa629/controllers/component_build_controller_pipeline.go#L404 . If we want to increase this, this will affect all tasks created in the pipelineRun , and we don't want that.

Already in touch with Konflux engineering to find a solution...